public/app/percona/ui-events/reducer.ts

```diff
@@ -0,0 +1,41 @@
+import { processDashboardEvents } from "app/percona/ui-events/events/dashboard";
+import { processFetchingEvents } from "app/percona/ui-events/events/fetching";
+import { processNotificationEvents } from "app/percona/ui-events/events/notification";
+
+const skipPrefixes = [
+  "@@",
+  "navIndex/",
+  "navBarTree/",
+  "panels/removePanels",
+  "appNotifications/hideAppNotification",
+];
+
+const shouldProcess = (type: string): boolean => {
+  if (!type) {
+    return false;
+  }
+
+  return !skipPrefixes.find(each => type.startsWith(each));
+};
+
+export interface Action {
+  payload: any;
+  type: string;
+}
+
+export const uiEventsReducer = (state: any = {}, action: Action) => {
+  try {
+    if (!shouldProcess(action.type)) {
```

Comment:
I plan to catch here settings set event, to identify if telemetry is disabled. On this event I will clean up the storage and disable processing.